source/simulator/parameters.cc

```diff
@@ -1949,6 +1959,14 @@ namespace aspect
                    ExcMessage ("The list of names for the mapped particle property fields needs to either be empty or have a length equal to "
                                "the number of compositional fields that are interpolated from particle properties."));
 
+      if (std::find(compositional_field_methods.begin(), compositional_field_methods.end(), AdvectionFieldMethod::fem_darcy_field)
+          != compositional_field_methods.end())
+        AssertThrow (std::find(names_of_compositional_fields.begin(),
+                               names_of_compositional_fields.end(),
+                               "porosity")
+                     != names_of_compositional_fields.end(),
+                     ExcMessage ("The Darcy advection field method only works if there is a compositional field named 'porosity'"));
```

Comment:
This is the first step, but you also need to check if the porosity field is actually advected with the darcy velocity as well.

Reply:
This is now resolved with the Assert below.